airbyte-integrations/connectors/source-gcs/unit_tests/test_source.py

```diff
@@ -0,0 +1,57 @@
+# Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+
+import re
+from pathlib import Path
+from unittest.mock import Mock
+
+import pytest
+from airbyte_cdk import AirbyteTracedException
+from airbyte_cdk.sources.file_based.availability_strategy import DefaultFileBasedAvailabilityStrategy
+from source_gcs import Config, Cursor, SourceGCS, SourceGCSStreamReader
+
+
+def _source_gcs(catalog, config):
+    return SourceGCS(
+        SourceGCSStreamReader(),
+        Config,
+        catalog,
+        config,
+        None,
+        cursor_cls=Cursor,
+    )
+
+
+def _catalog_path():
+    return Path(__file__).resolve().parent.joinpath("resource/catalog/")
+
+
+def _config_path():
+    return Path(__file__).resolve().parent.joinpath("resource/config/")
+
+
+def _mock_encoding_error():
+    DefaultFileBasedAvailabilityStrategy.check_availability_and_parsability = Mock(side_effect=AirbyteTracedException(message="encoding error"))
+
+
+def test_check_connection_with_bad_encoding(logger):
```

Comment:
from the source level, the error handling isn't specific to encoding, so let's make the test name generic